### PR TITLE
#279: Upgrade workflows

### LINKED ISSUE #279: Upgrade pipelines for cpp17

### PR DESCRIPTION
**Pull request by lifflander**
Fixes #279

### CODE REVIEW

File: .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 ############## Warning this is a generated file---do not modify ###############
 ###############################################################################
 
-name: PR tests (nvcc-11.0, ubuntu, mpich)
+name: PR tests (nvcc-11.4, ubuntu, mpich)
```
Comment (PhilMiller) [inline .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml]: Nvcc 11.4 is especially buggy. Go with 11.2

Comment (PhilMiller) [reply .github/workflows/dockerimage-nvcc-11-ubuntu-mpich.yml]: Or stick with 11.0

**Review by PhilMiller — CHANGES_REQUESTED** [review]

File: ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile [hunk 2]
```diff
@@ -1,10 +1,10 @@
 
-ARG cuda=10.1
+ARG cuda=11.4
```
Comment (PhilMiller) [inline ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile]: It looks like `${cuda}` is never used. Either drop it, or match 11.2 as the default.

**Review by PhilMiller — APPROVED** [review]
Please address the comment about `ARG cuda=11.4`. No need for me to re-review.

File: ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile (hunk 2 above)
Comment (lifflander) [reply ci/docker/ubuntu-20.04-nvidia-cpp.dockerfile]: Removed

**Pull request merged**